### PR TITLE
feat(ratelimit): Emit sys event detailing the rate limit configuration

### PR DESCRIPTION
**Pull request by tmessi**
A sys event will be emitted when the controller is constructed, and if
the rate limits are changed via a SIGHUP. The event contains the details
of the rate limits for each combination of resource and action. This
event can be used by operators to confirm the limits are set as they
would expect based on the controller config.

### CODE REVIEW

File: internal/daemon/controller/controller.go [hunk 1]
```diff
@@ -264,6 +264,7 @@ func New(ctx context.Context, conf *Config) (*Controller, error) {
 		return nil, fmt.Errorf("error initializing rate limiter: %w", err)
 	}
 	c.rateLimiter.Store(rateLimiter)
+	ratelimit.WriteLimitsSysEvent(c.baseContext, rateLimits, conf.RawConfig.Controller.ApiRateLimiterMaxEntries)
```
Comment (mgaffney) [inline internal/daemon/controller/controller.go:267]: `ratelimit.WriteLimitsSysEvent` returns an error. Can we handle the error? If not, can we `_ = ratelimit.WriteLimitsSysEvent` and add a comment to explain?

File: internal/daemon/controller/controller.go [hunk 2]
```diff
@@ -666,6 +667,7 @@ func (c *Controller) ReloadRateLimiter(newLimitConfigs ratelimit.Configs, newMax
 	}
 
 	old := c.rateLimiter.Swap(limiter)
+	ratelimit.WriteLimitsSysEvent(c.baseContext, ratelimits, newMaxEntries)
```
Comment (mgaffney) [inline internal/daemon/controller/controller.go:670]: Same as above

File: internal/ratelimit/limiter.go [hunk 3]
```diff
@@ -14,3 +19,63 @@ func NewLimiter(limits []*rate.Limit, maxEntries int) (*rate.Limiter, error) {
...
+
+type (
+	actionLimits    []limit
+	resourceActions map[string]actionLimits
+	resources       map[string]resourceActions
+)
+
+// WriteLimitsSysEvent writes a sys event that contains all of the provided
+// rate limits.
+func WriteLimitsSysEvent(ctx context.Context, limits []*rate.Limit, maxEntries int) error {
```
Comment (mgaffney) [inline internal/ratelimit/limiter.go:42]: When can this function return an error?

**Review by mgaffney — CHANGES_REQUESTED** [review]

File: internal/ratelimit/limiter.go (hunk 3 above)
Comment (tmessi) [reply internal/ratelimit/limiter.go:42]: Hmmm. I don't think it can. I'll update it.

**Comment by tmessi** [comment]
Closing in favor of #4089

**Pull request closed without merging**